targets/TARGET_NORDIC/TARGET_NRF5x/serial_api.c

```diff
@@ -1022,6 +1027,15 @@ void serial_init(serial_t *obj, PinName tx, PinName rx)
 
     uart_object->instance = instance;
 
+    /* Increment usage counter for this instance. */
+    nordic_nrf5_uart_state[instance].usage_counter++;
+
+    /* Enable instance on first usage. */
+    if (nordic_nrf5_uart_state[instance].usage_counter == 1) {
```

Comment:
Why does each peripheral instance have a usage counter? Shouldn't the usage counter be a part of the bank of peripherals?

Reply:
You enable/disable each instance separately.